src/js/lib.js

```diff
@@ -726,16 +709,15 @@ vjs.parseUrl = function(url) {
  * @private
  */
 function _logType(type, args){
-  var argsArray, noop, console;
-
   // convert args to an array to get array functions
-  argsArray = Array.prototype.slice.call(args);
+  let argsArray = Array.prototype.slice.call(args);
   // if there's no console then don't try to output messages
   // they will still be stored in vjs.log.history
   // Was setting these once outside of this function, but containing them
   // in the function makes it easier to test cases where console doesn't exist
-  noop = function(){};
-  console = window['console'] || {
+  let noop = function(){};
```

Comment:
I think it's totally reasonable to stick with the inline `function(){}` for readability sake. The performance argument (without backing evidence-- add _ten_ thousand massive grains of salt) is that the inline noop is a separate function instance, eats heap space, and could lead to a GC. It might not even show up in microbenchmarks like JSPerf because they probably don't trigger garbage collection during test runs but could be noticeable in a real app.

I don't want anybody headed to Google over that many "maybes". Let's stick with `function() {}` unless we have evidence a change is actually valuable.